test/parallel/test-stream-finished.js

```diff
@@ -396,3 +403,69 @@ testClosed((opts) => new Writable({ write() {}, ...opts }));
   r.destroyed = true;
   r.push(null);
 }
+
+{
+  // Regression https://github.com/nodejs/node/issues/33130
+
+  const response = new PassThrough();
+  response.write('chunk 1');
+  response.write('chunk 2');
+  response.write('chunk 3');
+  response.end();
+
+  class HelloWorld extends Duplex {
+    constructor(response) {
+      super();
+
+      this.response = response;
+      this.readMore = false;
+
+      response.once('end', () => {
+        this.push(null);
+      });
+
+      response.on('readable', () => {
+        if (this.readMore) {
+          this._read();
+        }
+      });
+    }
+
+    _read() {
+      const { response } = this;
+
+      this.readMore = true;
+
+      if (response.readableLength) {
+        this.readMore = false;
+      }
+
+      let data;
+      while ((data = response.read()) !== null) {
+        this.push(data);
+      }
+    }
+  }
+
+  const instance = new HelloWorld(response);
+  instance.end();
+
+  (async () => {
+    await new Promise((resolve) => setTimeout(resolve, 100));
```

Comment:
Is this line necessary?

Reply:
probably not, @szmarczak?

Reply:
We need to wait for the finish event so `instance.finished` is true

Reply:
But it can be done faster

Reply:
`await once(instance, 'finish');` should do that without a timeout, no?

Reply:
Why do we need to wait for finish?

Reply:
@szmarczak: How about now, still good?